ert/data/_record.py

```diff
@@ -351,10 +354,11 @@ async def _transmit_blob_record(self, record: BlobRecord) -> str:
     async def _load_numerical_record(self) -> NumericalRecord:
         async with aiofiles.open(str(self._uri), mode="rt", encoding="utf-8") as f:
             contents = await f.read()
+        serializer = get_serializer(SharedDiskRecordTransmitter._INTERNAL_MIME_TYPE)
         if self._record_type == RecordType.MAPPING_INT_FLOAT:
-            data = json.loads(contents, object_hook=parse_json_key_as_int)
+            data = serializer.decode(contents)
         else:
-            data = json.loads(contents)
+            data = serializer.decode(contents)
```

Comment:
Unnecessary branching? Both cases are equal. They didn't used to be, but you mentioned you made an issue for that. Could you add that to a `# TODO`?

Reply:
Is YAML respecting the mapping from int to float?

Reply:
Yes, forgot to clean this up now that it was switched to yaml! And yes yaml does not have the same problem as JSON, as there is a difference between the key `"10"` and `10`:

```
>>> yaml.load(yaml.dump({10:10}))
{10: 10}
>>> yaml.load(yaml.dump({"10":10}))
{'10': 10}
```